doc/voq/voq_hld.md

```diff
@@ -0,0 +1,879 @@
+      "Ethernet2": {},
+      "Ethernet3": {},
+      "Ethernet1"|"10.0.0.1/16": {},
+      "Ethernet2"|"20.0.0.1/16": {},
+      "Ethernet3"|"30.0.0.1/16": {}
+   }
+}
+```
+
+##### In ASIC# 1
+
+```
+{
+   "INTERFACE": {
+      "Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {},
+      "Ethernet128"|"10.1.0.1/16": {},
+      "Ethernet129"|"20.1.0.1/16": {},
+      "Ethernet130"|"30.1.0.1/16": {}
+   }
+}
+```
+### Interface infomation for local and remote system ports in APP DB
+IP address configuration for remote system ports are done in local config DB as shown above
+
+The interface information for remote system ports are populated by VoqDbSyncd
+
+#### APP DB Objects:
+
+##### In ASIC #0
+
+```
+{
+   "INTF_TABLE": {
+      "Ethernet1": {},
+      "Ethernet2": {},
+      "Ethernet3": {},
+      "Ethernet1":"10.0.0.1/24": {},
+      "Ethernet2":"20.0.0.1/24": {},
+      "Ethernet3":"30.0.0.1/24": {},
+      "Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {}
+   }
+}
+```
+##### In ASIC #1
+
+```
+{
+   "INTF_TABLE": {
+      "Ethernet1": {},
+      "Ethernet2": {},
+      "Ethernet3": {},
+       Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {},
+      "Ethernet128":"10.1.0.1/24": {},
+      "Ethernet129":"20.1.0.1/24": {},
+      "Ethernet130":"30.1.0.1/24": {}
+   }
+}
+```
+
+### Static Neighbor configuration in Config DB
+This is optional. 
+
+#### Config DB objects:
+Static neighbors are configured on local ports
+
+##### On ASIC #0
+
+```
+{
+   "NEIGH": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+##### On ASIC #1
+
+```
+{
+   "NEIGH": {
+      "Ethernet128:10.1.0.2": {
+         "neigh": "02:06:1a:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+### Neighbor information in APP DB
+
+#### APP DB Objects:
+The NEIGH_TABLE has entries for locally learned neighbors on local ports
+
+The NEIGH_TABLE also has entries for neighbors learned/configured on remote system ports (populated by VoqDbSyncd)
+
+##### On ASIC #0
+```
+{
+  "NEIGH_TABLE": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "family", "IPv4",
+         "vrf": "1"
+      },
+      "Ethernet2:20.0.0.2": {
+         "neigh": "02:06:0b:00:00:01",
+         "family", "IPv4",
+         "vrf": "1"
+      },
+      "Ethernet128:10.1.0.1": {
+         "neigh": "02:16:0a:00:00:01",
+         "family": "IPv4",
+         "encap_index": "4097",
+         "vrf": "1",
+      }
+  }
+}
+```
+
+##### On ASIC #1
+```
+{
+  "NEIGH_TABLE": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "family": "IPv4"
+         "encap_index": "4096",
+         "vrf": "1"
+      },
+      "Ethernet128:10.1.0.2": {
+         "neigh": "02:16:0a:00:00:01",
+         "vrf": "1"
+      },
+      "Ethernet129:20.1.0.2": {
+         "neigh": "0b2:16:0b:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+
```

Comment:
The testing section from the Arch document can be migrated here because it is more relevant. We can also include other tests that we have specified for route/neighbor creation here.

Reply:
You can refer to the Arch document section and add something along the lines of

## 3.2 Distributed VoQ functionality

### 3.2.3 test_virtual_chassis.py

Is the top level test driver that executes testcases against the virtual chassis.

Sysport handling is tested by test_chassis_sysport which validates that
* System ports can be populated in CONFIG_DB
* Orchagent programs the correct SAI Redis ASIC_DB state to represent the configured sysport in a multi-instance environment

Reply:
It might be better if you can add this part to the document. Let me know